### PR TITLE
PIM-6876: escape u001f character to workaround a mysql bug

### PR DESCRIPTION
**Pull request by skeleton**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added Behats                      | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/CreateProductIntegration.php [hunk 1]
```diff
@@ -979,6 +979,37 @@ public function testResponseWhenSettingProductValueWithAnUnknownAttribute()
...
+        $client->request('POST', '/api/rest/v1/products', [], [], [], $data);
+        $response = $client->getResponse();
+        $this->assertSame(Response::HTTP_CREATED, $response->getStatusCode());
+        $this->assertEmpty($response->getContent());
+
+        $this->get('akeneo_elasticsearch.client.product')->refreshIndex();
+
+        $client->request('GET', '/api/rest/v1/products/foo');
+        $response = $client->getResponse();
+        $this->assertSame(Response::HTTP_OK, $response->getStatusCode());
```
Comment (ahocquard) [inline src/Pim/Bundle/ApiBundle/tests/integration/Controller/Product/CreateProductIntegration.php]: check the return value as well to be sure that you get the character.

File: src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/DBAL/Types/JsonType.php [hunk 2]
```diff
@@ -29,6 +29,8 @@ public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $pla
      */
     public function convertToPHPValue($value, AbstractPlatform $platform)
     {
+        $value = preg_replace('/[\x00-\x1F]/', '\\u001f', $value);
```
Comment (jjanvier) [inline src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/DBAL/Types/JsonType.php]: could be cool to add a link to the MySQL bug

Comment (ahocquard) [reply src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/DBAL/Types/JsonType.php]: And it's fixed in Mysql 8.0 no ? We could add a comment : `fixed in Mysql 8.0` to think about deleting it when we will upgrade

Comment (SamirBoulil) [reply src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/DBAL/Types/JsonType.php]: bug link: https://bugs.mysql.com/bug.php?id=87722

Comment (SamirBoulil) [reply src/Akeneo/Bundle/StorageUtilsBundle/Doctrine/DBAL/Types/JsonType.php]: fixed in 8.0 along with this bug: https://bugs.mysql.com/bug.php?id=86898 :(

**Pull request merged**